### PR TITLE
Issue #13100: Create tests from AbbreviationAsWordInName xdoc examples

### PR DESCRIPTION
**Pull request by stoyanK7**
Part of #13100

https://checkstyle.sourceforge.io/config_naming.html#AbbreviationAsWordInName

### CODE REVIEW

**Comment by nrmancuso** [comment]
> Should we just remove the call or suppress it?

ForbiddenAPIs should not be looking at test code at all, right?

**Comment by stoyanK7** [comment]
@nrmancuso  Done. Extracted commit at https://github.com/checkstyle/checkstyle/pull/13215. I need it merged because other transfers have the same issue.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example7.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.naming.abbreviationaswordinname;
+
+class Example7 {
+  int counterXYZ = 1;
+  final int customerID = 2;
+  static int nextID = 3;
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example7.java:21]: Violations are expected, there is mistake in config, length= 0

**Review by romani — CHANGES_REQUESTED** [review]
Recheck config one time

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example7.java (hunk 1 above)
Comment (stoyanK7) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example7.java:21]: Are you sure? For me it looks okay.
```xml
<module name="AbbreviationAsWordInName">
    <property name="allowedAbbreviations" value="ALLOWED"/>
    <property name="ignoreStaticFinal" value="false"/>
</module>
```
```java
public class MyClass {
    public int counterXYZ = 1;                // OK
    public final int customerID = 2;          // OK
    public static int nextID = 3;             // OK
    public static final int MAX_ALLOWED = 4;  // OK, abbreviation is allowed
}
```
Screenshot from 2023-06-14 09-21-09

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example7.java:21]: You are right, I looked at 6th example

**Review by romani — APPROVED** [review]
Ok to merge

**Review by nrmancuso — APPROVED** [review]

**Pull request merged**